File: contests/FIBEASY.cpp
#include<bits/stdc++.h>
#include <utility> 
#define ll long long int
ll maxx = 1e14;
using namespace std;
int main()
{
    ll fib[310];
    fib[0]=0;
    fib[1]=1;
    for(int i=2;i<300;i++)
    {
        fib[i]=(fib[i-1]+fib[i-2])%10;
    }
    ll t;
    cin>>t;
    while(t--)
    {
        ll n,j=2;
        cin>>n;
       // cout<<(1<<j)<<endl;
        // while(n>=(1LL << j))
        // {
        //     j++;
        // }
   //    cout<<j<<endl;

        
        ll ind=(1<<(j-1));
        cout<<fib[(ind-1)%300]<<endl;

    }
}

File: contests/FIBEASY.py


fib=[]
fib[0]=0
fib[1]=1

for i in range(2,300):
    fib[i]=(fib[i-1]+fib[i-2])/10

t=int(input())

for x in range(0,t):
    n=int(input())
    j=1
    while(n>=(1<<j))
    j=j+1
    ind=1<<(j-1)
    print(fib[ind%300])